### ISSUE TITLE
cusparse_build_version outputs version greater than latest cusparse version

### ISSUE THREAD

**Issue by djhavert** [google/jax#20079]
### Description

Hi There,

I'm trying to run JAX on a fresh VM on cloud infrastructure. The error is on the first line when printing the environment info
```
CUDA backend failed to initialize: Found cuSPARSE version 12103, but JAX was built against version 12200, which is newer. The copy of cuSPARSE that is installed must be at least as new as the version against which JAX was built. (Set TF_CPP_MIN_LOG_LEVEL=0 and rerun for more info.)
```

This seems like a bug in the code that checks the cuSPARSE version because the [latest version of cuSPARSE](https://pypi.org/project/nvidia-cusparse-cu12/) is `12.2.0.103`. 

As seen on my system, I have this version:
```
$ python3 -m pip list | grep cusparse
nvidia-cusparse-cu12     12.2.0.103
```

And using the jax function after opening a `python3` shell, you can see it finds this version, but cuts off the middle section.
```
>>> from jax._src.lib import cuda_versions
>>> cuda_versions.cusparse_get_version()
12103
```
Should this instead read 1220103?

The build version expected by JAX is somehow larger than this.
```
>>> from jax._src.lib import cuda_versions
>>> cuda_versions.cusparse_build_version()
12200
```
I suspect this is supposed to read `1220000` instead? Either way, this version as currently depicted doesn't seem to exist and is definitely larger than the most recent cuSPARSE version available with pip, so this seems like a bug in the `cuda_versions.cusparse_build_version()` function.

If there is anything else I can provide to help track this down, let me know. I'm trying to help some researchers get JAX running on GPUs on Jetstream2, and I cannot get past this roadblock.

### System info (python version, jaxlib version, accelerator, etc.)

```
>>> import jax; jax.print_environment_info()
CUDA backend failed to initialize: Found cuSPARSE version 12103, but JAX was built against version 12200, which is newer. The copy of cuSPARSE that is installed must be at least as new as the version against which JAX was built. (Set TF_CPP_MIN_LOG_LEVEL=0 and rerun for more info.)
jax:    0.4.25
jaxlib: 0.4.25
numpy:  1.26.4
python: 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0]
jax.devices (1 total, 1 local): [CpuDevice(id=0)]
process_count: 1
platform: uname_result(system='Linux', node='eagerly-healthy-wallaby', release='5.15.0-97-generic', version='#107-Ubuntu SMP Wed Feb 7 13:26:48 UTC 2024', machine='x86_64')


$ nvidia-smi
Tue Mar  5 18:40:36 2024
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 535.129.03             Driver Version: 535.129.03   CUDA Version: 12.2     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  GRID A100X-40C                 On  | 00000000:04:00.0 Off |                    0 |
| N/A   N/A    P0              N/A /  N/A |      0MiB / 40960MiB |      0%      Default |
|                                         |                      |             Disabled |
+-----------------------------------------+----------------------+----------------------+

+---------------------------------------------------------------------------------------+
| Processes:                                                                            |
|  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
|        ID   ID                                                             Usage      |
|=======================================================================================|
|  No running processes found                                                           |
+---------------------------------------------------------------------------------------+
```

**Comment by jakevdp** [comment]
JAX's `cusparse_version` just returns `CUSPARSE_VERSION` as defined in the cusparse C headers. The definition looks something like this depending on which CUDA version you're using:
```
#define CUSPARSE_VER_MAJOR 12
#define CUSPARSE_VER_MINOR 1
#define CUSPARSE_VER_PATCH 3
#define CUSPARSE_VER_BUILD 153
#define CUSPARSE_VERSION (CUSPARSE_VER_MAJOR * 1000 + \
                          CUSPARSE_VER_MINOR *  100 + \
                          CUSPARSE_VER_PATCH)
```
This would become `12103`. `12200` would come from version `12.2.0`. If JAX is reporting a different CUDA version than in header files that you see on your system, then it is likely accessing a different CUDA installation than you are expecting. For example, if you install pytorch via `pip`, it will bundle its own CUDA headers that may be higher up in your `LD_LIBRARY_PATH` than the other headers you're looking at.

Does that help answer your question?

**Comment by djhavert** [comment]
Ah, I see what you mean. I was confused because the `103` of `12103` happened to match the `103` of `12.2.0.103`. It turns out these are not correlated. So even though the latest version of cusparse I have through pip is `12.2.0.1031`, jax must be pulling the cusparse version from elsewhere. Likely I have cusparse `12.1.3.153` installed somewhere, though where I'm not sure yet. I'll go digging for that and see if I can remove it from the path.

Confirmed. Cusparse was installed elsewhere. I removed it from LD_LIBRARY_PATH and all is good now.
